content/docs/concepts/architecture/index.md

```diff
@@ -140,21 +145,128 @@ Any of these services can be replaced by a service for another tool
 to create a Keptn-service that integrates other tools.
+
+### Summary of NATS behavior for Keptn on a single cluster
+
+On a single-cluster Keptn instance,
+the Keptn control plane and execution plane are both installed on the same cluster
+and they communicate using NATS.
+Execution plane services have a distributor pod
+that subscribes to and publishes events on behalf of the execution plane service.
+
+Environment variables documented
+on the [distributor](../../0.19.x/reference/miscellaneous/distributor) reference page
+control how the distributor behaves,
+including setting the `PUBSUB_URL` environment variable that the distributor uses to locate the NATS cluster.
+
+The flow can be summarized as follows.
+Note that this discussion assumes using `helm-service` and tasks like `deployment`
+but another service could be used for this processing
+and any tool could listen for tasks with names other than those of the standard tasks
+that are documented on the [shipyard](../../0.19.x/reference/files/shipyard/#fields) reference page.
+
+1. The distributor for the execution plane services on a control plane
+   handles the subscriptions and publishes operations for the execution plane service
+   by subscribing to NATS subjects that Keptn creates dynamically.
+
+   For sequence-level events:
+
+   ```
+   sh.keptn.event.<stage>-<sequence>.<verb>
+   ```
+
+   For example, the control plane might publish the following to the subject:
+
+   ```
+   sh.kept.event.dev.delivery.triggered
+   sh.keptn.event.deployment.triggered
+   ```
+
+   For task-level events:
+
+   ```
+   .sh.keptn.event.<task>.<verb>
+   ```
+
+   Optionally, the control plain can send one or more events in the form:
+
+   ```
+   sh.keptn.<task>.status.changed
+   ```
+   This is useful to signal status updates during long-running tasks.
+ 
+1. The Helm-Service Distributor (HSD) subscribes to the subject:
+
+   ```
+   sh.keptn.event.deployment
+   ```
+   And receives:
+
+   ```
+   sh.keptn.event.deployment.started
+   ```
+   as well as the JSON event body.
+
+1. HSD triggers the helm-service and publishes to the subject:
+
+   ```
+   sh.keptn.vent.deployment.started
+   ```
+   as well as the JSON event body.
+
+
+1. The helm-service finishes the deployment and the HSD publishess to the subject:
+
+   ```
+   sh.keptn.event.deployment.finished
+   ```
+   as well as the JSON event body.
+
+### Summary of behavior of NATS for Keptn on a multi-cluster instance:
+
+In a [multi-cluster](../../install/multi-cluster) configuration,
+an execution plane is a namespace or cluster other than where the control plane runs
+that runs a Keptn service.
+The distributor was originally designed to work for both the control plane and the remote execution planes
+but, for recent releases, most execution plane services use
+[go-sdk](https://github.com/stellar/go/blob/master/docs/reference/readme.md) rather than the distributor.
```

Comment:
Maybe this info should be more prominent further up the file? All newer services and most keptn internal service use the go-sdk now.

Reply:
Sounds good!